### PR TITLE
For #20871 Mark strings for removal

### PR DESCRIPTION
**Pull request by Amejia481**
Related https://github.com/mozilla-mobile/fenix/wiki/Remove-strings/ and https://github.com/mozilla-mobile/fenix/issues/20871
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1483,7 +1483,7 @@
     <!-- Preference option for never saving passwords in Fenix -->
     <string name="preferences_passwords_save_logins_never_save">Never save</string>
     <!-- Preference for autofilling saved logins in Fenix -->
-    <string name="preferences_passwords_autofill" tools:ignore="UnusedResources">Autofill</string>
+    <string name="preferences_passwords_autofill" moz:removedIn="93" tools:ignore="UnusedResources">Autofill</string>
```
Comment (MozillaNoah) [inline app/src/main/res/values/strings.xml:1486]: will strings ever be removed in patches and/or dot-version releases?

Comment (Amejia481) [reply app/src/main/res/values/strings.xml:1486]: Ideally our automatization should take care of when we do release, hopefully we can integrate it to our release process.

Comment (MozillaNoah) [reply app/src/main/res/values/strings.xml:1486]: Sorry, I was more asking if we have ever had patch releases that removed strings, or if that's typically reserved for major version releases.

Comment (Amejia481) [reply app/src/main/res/values/strings.xml:1486]: I think ideally we would like to run removing string tool for major release to make sure the removed strings are not longer used as having minor and bug fixed versioning could bring more complexity.

Comment (Amejia481) [reply app/src/main/res/values/strings.xml:1486]: At least, this what see initially, do you have any suggestions ? :)

**Review by MozillaNoah — APPROVED** [review]

**Pull request merged**